textworld/gym/envs/batch_env.py

```diff
@@ -164,24 +160,13 @@ def step(self, actions: List[str]) -> Tuple[List[str], int, bool, Dict[str, List
                 env.call("step", action)
                 results.append(None)
 
-        obs, rewards, dones, infos = [], [], [], None
-        for i, (env, result) in enumerate(zip(self.envs, results)):
-            if result is None:
-                result = env.result()
-
-            ob, reward, done, info = result
-
-            obs.append(ob)
-            rewards.append(reward)
-            dones.append(done)
-
-            if infos is None:
-                infos = {k: [] for k in info}
-
-            for k, v in info.items():
-                infos[k].append(v)
+        results = [result if result else env.result() for env, result in zip(self.envs, results)]
```

Comment:
You can do `result or env.result() for ...`

Reply:
Good point.